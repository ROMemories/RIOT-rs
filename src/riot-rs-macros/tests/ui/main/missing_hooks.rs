#![no_main]
#![feature(type_alias_impl_trait)]
#![feature(used_with_arg)]

use riot_rs::define_peripherals;

// As the macro will fail, this import will not get used
#[allow(unused_imports)]
use riot_rs::embassy::usb::UsbBuilderHook;

define_peripherals!(Peripherals {});

#[riot_rs::main]
async fn main(_peripherals: Peripherals, _usb_builder_hook: UsbBuilderHook) {}
