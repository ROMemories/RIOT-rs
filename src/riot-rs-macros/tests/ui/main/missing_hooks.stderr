error[E0061]: this function takes 2 arguments but 1 argument was supplied
  --> tests/ui/main/missing_hooks.rs:14:10
   |
13 | #[riot_rs::main]
   | ---------------- an argument of type `&'static Delegate<embassy_usb::builder::Builder<'static, UsbDriver>>` is missing
14 | async fn main(_peripherals: Peripherals, _usb_builder_hook: UsbBuilderHook) {}
   |          ^^^^
   |
note: function defined here
  --> tests/ui/main/missing_hooks.rs:14:10
   |
14 | async fn main(_peripherals: Peripherals, _usb_builder_hook: UsbBuilderHook) {}
   |          ^^^^ -------------------------  ---------------------------------
help: provide the argument
   |
13 | main(#[riot_rs::main], /* &'static Delegate<embassy_usb::builder::Builder<'static, UsbDriver>> */)
   |
